### PR TITLE
build: mac deployment unification

### PR DESCRIPTION
**Pull request by fanquake**
This consolidates our macOS build code so that `.DS_Store` generation is the same when running `make deploy` for macOS when building on Linux and macOS, rather than maintaining two version of code that essentially do the same thing (just slightly differently).

It also removes unused code and any AppleScript usage, automates finding translation files and generally simplifies `macdeployqtplus`. It also gets rid of the annoying "popping up" behaviour during DMG generation, names the created image `Bitcoin-Core.dmg` rather than `Bitcoin-Qt.dmg`.

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK

**Comment by laanwj** [comment]
> One outstanding issue is that there is either a bug, or a backwards incompatibility with the latest (2.1.0) version of mac_alias:

It's a bug. Looks like they tried to spread the expression over multiple lines but forgot to either add parenthesis around the tuple or `\`s.

https://github.com/al45tair/mac_alias/blob/master/mac_alias/alias.py#L282

**Comment by RandyMcMillan** [comment]
Concept ACK!

**Comment by kizoojun01** [comment]
Kizoojun

**Comment by fanquake** [comment]
> Edit: created an upstream issue al45tair/mac_alias#9

Great, thanks. The upstream issue [has been resolved](https://github.com/al45tair/mac_alias/commit/adee16b957e3ff83618b6f28ec372caf4bf0ac35) with the release of mac_alias 2.1.1. I've added an additional commit to this PR to switch to that version in depends. I've also updated the PR description to remove no-longer-relevant info.

This PR should now be ready for review.

File: contrib/macdeploy/macdeployqtplus [hunk 1]
```diff
@@ -16,9 +16,13 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
-import subprocess, sys, re, os, shutil, stat, os.path, time
-from string import Template
+import plistlib
```
Comment (dergoegge) [inline contrib/macdeploy/macdeployqtplus:19]: This import was only used by the `-fancy` option and could also be removed since it is now unused.

**Review by dergoegge — APPROVED** [review]
ACK b685f60a08007e0ae8a5564ee68cd94f9015d899 - Less and cleaner code looks good. I tested this with `make deploy` and everything still works + the popup during DMG generation is gone.

**Review by willcl-ark — COMMENTED** [review]
utACK. I don't have an Apple Developer account so can't get the SDK to test with I'm afraid.

File: contrib/macdeploy/macdeployqtplus (hunk 1 above)
Comment (fanquake) [reply contrib/macdeploy/macdeployqtplus:19]: Good catch. However I'm going to merge this now, and we'll take care of the leftover import shortly.

**Comment by Sjors** [comment]
We should add an instruction before `make deploy` to install the Python dependency:

```
pip3 install ds_store
```

**Comment by hebasto** [comment]
It appears this PR breaks `make deploy` on macOS 10.14.6 (18G9323):
```
$ make deploy
...
created: /Users/hebasto/bitcoin/Bitcoin-Core.temp.dmg
+ Applying fancy settings +
+ Finalizing .dmg disk image +
"disk23" ejected.
hdiutil: convert failed - Resource temporarily unavailable
Traceback (most recent call last):
  File "/Users/hebasto/bitcoin/./contrib/macdeploy/macdeployqtplus", line 687, in <module>
    run(["hdiutil", "convert", tempname, "-format", "UDZO", "-o", appname, "-imagekey", "zlib-level=9"], check=True, universal_newlines=True)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['hdiutil', 'convert', 'Bitcoin-Core.temp.dmg', '-format', 'UDZO', '-o', 'Bitcoin-Core', '-imagekey', 'zlib-level=9']' returned non-zero exit status 1.
make: *** [Makefile:1282: Bitcoin-Core.dmg] Error 1
```

Newer macOS versions—10.15 and 11.x—are not affected.

**Comment by fanquake** [comment]
> hdiutil: convert failed - Resource temporarily unavailable

This looks like an intermittent issue, and I would be somewhat surprised if the changes here worked on macOS 10.15 but not 10.14. I don't think there's anything we're doing that would be specific to the OS, or a certain version of a tool, like `hdiutil` etc.

Does `make deploy` work on 10.14 with this change reverted? Without any more info there's not a lot to go on here.

**Comment by hebasto** [comment]
> Does `make deploy` work on 10.14 with this change reverted?

Yes, it does.

I think @jarolrod could also confirm such behavior?

**Pull request merged**